src/xmipp/libraries/reconstruction/classify_compare_classes.cpp

```diff
@@ -60,26 +60,26 @@ void ProgCompareClass::show()
 
 void ProgCompareClass::run()
 {
-	MetaData MD1(formatString("classes@%s",fnClass1.c_str()));
-	MetaData MD2(formatString("classes@%s",fnClass2.c_str()));
-        std::vector<int> ref1, ref2;
-        int aux;
-        FOR_ALL_OBJECTS_IN_METADATA(MD1)
-        {
-            MD1.getValue(MDL_REF,aux,__iter.objId);
-            ref1.push_back(aux);
-        }
-        FOR_ALL_OBJECTS_IN_METADATA(MD2)
-        {
-            MD2.getValue(MDL_REF,aux,__iter.objId);
-            ref2.push_back(aux);
-        }
+	MetaDataVec MD1(formatString("classes@%s",fnClass1.c_str()));
+	MetaDataVec MD2(formatString("classes@%s",fnClass2.c_str()));
+    std::vector<int> ref1, ref2;
```

Comment:
you mind to use here the method which reads the entire column at once?

Reply:
TODO read entire column once.